python/cudf/cudf/_lib/pylibcudf/gpumemoryview.pyx

```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+
+
+cdef class gpumemoryview:
+    """Minimal representation of a memory buffer."""
+    def __init__(self, object obj):
+        try:
+            cai = obj.__cuda_array_interface__
+        except AttributeError:
+            raise ValueError(
+                "gpumemoryview must be constructed from an object supporting "
+                "the CUDA array interface"
+            )
+        self.obj = obj
+        # TODO: Need to respect readonly
+        self.ptr = cai["data"][0]
+
+    @property
+    def __cuda_array_interface__(self):
+        return self.obj.__cuda_array_interface__
```

Comment:
If you're effectively asking whether we could cache this, I think the answer is yes.

Reply:
Now using `functools.cached_property`.

Reply:
Can this bite us if the pointer gets changed from underneath us? I'm thinking situations like when `obj` is a `Series` and the column beneath that Series is changed.

Reply:
IIUC that sequence would require creating a new `gpumemoryview` from the new `Column` before any pylibcudf algorithms would be called, right?

Reply:
We discussed this offline and came to two conclusions:
1. Caching this property should be safe. The case that would be problematic would be if the `gpumemoryview` was holding on to a reference to a `Series` or other higher-level object instead of the underlying `Buffer` since the data underneath any cudf `Column` could be swapped out by some operations. The current code always converts from a `Buffer` though, so that conversion is safe.
2. `cached_property` is [actually broken in Cython code](https://github.com/cython/cython/issues/5523).

Therefore, while it may be safe to cache this value, it isn't actually possible with the current Cython, so I've removed this.